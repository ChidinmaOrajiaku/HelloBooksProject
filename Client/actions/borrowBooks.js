import axios from 'axios';
import { BORROW_BOOKS_SUCCESSFUL, BORROW_BOOKS_FAILED } from './types';


/**
 * Borrowed books response if successful
 * @export borrowBooksResponse
 * @param {object} response
 * @returns {object} of borrowed books response
 */
export function borrowBooksResponse(response) {
  return {
    type: BORROW_BOOKS_SUCCESSFUL,
    response
  };
}

/**
   * Borrowed books response if failed
   * @export borrowBooksError
   * @param {object} error
   * @returns {object} of borrowed books error
   */
export function borrowBooksError(error) {
  return {
    type: BORROW_BOOKS_FAILED,
    error
  };
}

export const borrowRequest = (userId, booksId) => dispatch => axios.post(`/api/v1/users/${userId}/books`, booksId)
  .then((res) => {
    dispatch(borrowBooksResponse(res.data));
  }).catch((error) => {
    dispatch(borrowBooksError('An error occurred'));
  });
