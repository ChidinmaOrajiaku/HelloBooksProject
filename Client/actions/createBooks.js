import axios from 'axios';
import { CREATE_BOOKS_SUCCESSFUL, CREATE_BOOKS_FAILED } from './types';

/**
 *
 * Create books response if successful
 * @export createBooksResponse
 * @param {object} response
 * @returns {object} create books response
 */
export function createBooksResponse(response) {
  return {
    type: CREATE_BOOKS_SUCCESSFUL,
    response
  };
}

/**
   * Create books error if not successful
   * @export createBooksError
   * @param {object} error
   * @returns {object} of create book error
   */
export function createBooksError(error) {
  return {
    type: CREATE_BOOKS_FAILED,
    error
  };
}

export const adminAddRequest = bookData => dispatch => axios.post('/api/v1/users/books', bookData)
  .then((res) => {
    dispatch(createBooksResponse(res.data));
  }).catch((error) => {
    dispatch(createBooksError(error));
  });
