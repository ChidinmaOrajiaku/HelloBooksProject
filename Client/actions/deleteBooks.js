import axios from 'axios';
import { DELETE_BOOKS_SUCCESSFUL, DELETE_BOOKS_FAILED } from './types';

/**
 * Delete books response if successful
 * @export
 * @param {object} response
 * @returns {object} of delete books response
 */
export function deleteBooksResponse(response) {
  return {
    type: DELETE_BOOKS_SUCCESSFUL,
    response
  };
}

/**
   * Delete books error if not successful
   * @export
   * @param {object} error
   * @returns {object} of delete books error
   */
export function deleteBooksError(error) {
  return {
    type: DELETE_BOOKS_FAILED,
    error
  };
}

export const adminDeleteRequest = bookId => dispatch => axios.delete(`/api/v1/books/${bookId}`)
  .then((res) => {
    dispatch(deleteBooksResponse(res.data));
  }).catch((error) => {
    dispatch(deleteBooksError('An error occurred'));
  });
