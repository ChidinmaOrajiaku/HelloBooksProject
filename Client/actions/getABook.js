import axios from 'axios';
import { GET_A_BOOK_SUCCESSFUL, GET_A_BOOK_FAILED, GET_A_BOOK_REQUEST } from './types';


/**
   * 
   * 
   * @export
   * @param {any} image 
   * @returns 
   */
export function getABookRequest(data) {
  return {
    type: GET_A_BOOK_REQUEST,
    data
  };
}

/**
 * 
 * 
 * @export
 * @param {any} image 
 * @returns
 */
export function getABookResponse(response) {
  return {
    type: GET_A_BOOK_SUCCESSFUL,
    response
  };
}

/**
   * 
   * 
   * @export
   * @param {any} image 
   * @returns 
   */
export function getABookError(error) {
  return {
    type: GET_A_BOOK_FAILED,
    error
  };
}

export const getBookRequest = id => dispatch => axios.get(`/api/v1/books/${id}`).then((res) => {
  localStorage.getItem('jwtToken');
  dispatch(getABookResponse(res.data));
}).catch((error) => {
  dispatch(getABookError(error));
});
