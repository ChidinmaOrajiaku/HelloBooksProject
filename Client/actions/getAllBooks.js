import axios from 'axios';
import { GET_BOOKS_SUCCESSFUL, GET_BOOKS_FAILED, GET_BOOKS_REQUEST } from './types';


/**
   *
   *
   * @export
   * @param {any} data
   * @returns {object} get books reequest data
   */
export function getBooksRequest(data) {
  return {
    type: GET_BOOKS_REQUEST,
    data
  };
}

/**
 *
 *
 * @export
 * @param {any} response
 * @returns {object} gets a response when request is successful
 */
export function getBooksResponse(response) {
  return {
    type: GET_BOOKS_SUCCESSFUL,
    response
  };
}

/**
   *
   *
   * @export
   * @param {any} error
   * @returns {object} gets an error when request fails
   */
export function getBooksError(error) {
  return {
    type: GET_BOOKS_FAILED,
    error
  };
}

export const getRequest = () => dispatch => axios.get('/api/v1/users/books')
  .then((res) => {
    dispatch(getBooksResponse(res.data));
  }).catch((error) => {
    dispatch(getBooksError('An error occurred'));
  });
