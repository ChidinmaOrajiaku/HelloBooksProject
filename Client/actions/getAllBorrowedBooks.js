import axios from 'axios';
import { GET_BORROWED_BOOKS_SUCCESSFUL, GET_BORROWED_BOOKS_FAILED } from './types';


/**
 * Get borrowed books response if successful
 * @export getBorrowedBooksResponse
 *
 * @param {object} response
 *
 * @returns {object} of borrowed books response when request is successful
 */
export function getBorrowedBooksResponse(response) {
  return {
    type: GET_BORROWED_BOOKS_SUCCESSFUL,
    response
  };
}

/**
   * Get books error if not successful
   * @export getBorrowedBooksError
   *
   * @param {object} error
   *
   * @returns {object} of borrowed books response when request fails
   */
export function getBorrowedBooksError(error) {
  return {
    type: GET_BORROWED_BOOKS_FAILED,
    error
  };
}

export const admingetBorrowedRequest = () => dispatch => axios.get('/api/v1/users/books/unreturned')
  .then((res) => {
    dispatch(getBorrowedBooksResponse(res.data));
  }).catch((error) => {
    dispatch(getBorrowedBooksError('An error occurred'));
  });
