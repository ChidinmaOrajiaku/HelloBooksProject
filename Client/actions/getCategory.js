import axios from 'axios';
import { GET_CATEGORY_SUCCESSFUL, GET_CATEGORY_FAILED, GET_CATEGORY_REQUEST } from './types';


/**
   * 
   * 
   * @export
   * @param {any} data 
   * @returns {data} category
   */
export function getCategoryRequest(data) {
  return {
    type: GET_CATEGORY_FAILED,
    data
  };
}

/**
   * 
   * 
   * @export
   * @param {any} response
   * @returns {response} category
   */
export function getCategoryResponse(response) {
  return {
    type: GET_CATEGORY_SUCCESSFUL,
    response
  };
}

/**
   * 
   * 
   * @export
   * @param {any} error 
   * @returns {error} error
   */
export function getCategoryError(error) {
  return {
    type: GET_CATEGORY_REQUEST,
    error
  };
}

export const getAllCategoryRequest = () => dispatch => axios.get('/api/v1/books/category/all').then((res) => {
  localStorage.getItem('jwtToken');
  dispatch(getCategoryResponse(res.data));
}).catch((error) => {
  dispatch(getCategoryError(error));
});
