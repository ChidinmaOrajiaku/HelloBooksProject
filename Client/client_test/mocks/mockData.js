import { updatePassword } from "../../actions/updatePassword";

const mockData = {
  userDetails: {
    message: 'Welcome admin',
    username: 'admin',
    token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6ImRhZGR5Iiwicm9sZSI6InVzZXIiLCJpZCI6MiwiaWF0IjoxNTEzODc1ODE3LCJleHAiOjE1MTM5NjIyMTd9.oexUFhi08vhMU88LmZJvRDy9JFdThMpfb3Qz9DWWzgk'
  },
  userDetailsResponse: {
    username: 'daddy',
    role: 'user',
    id: 2,
    iat: 1513875817,
    exp: 1513962217
  },
  userCountResponse: {
    count: 7
  },
  borrowBookData: {
    bookId: 1,
    userId: 1,
  },
  borrowBookDataResponse: {
    returned: false,
    id: 24,
    userId: 1,
    bookId: 7,
    toReturnDate: '2018-01-14',
    updatedAt: '2017-12-21T21:16:08.442Z',
    createdAt: '2017-12-21T21:16:08.442Z',
    returnDate: null,
    categoryId: null
  },
  createBookData: {
    title: 'a',
    author: 'a',
    category: 'a',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1511133007/g2exfx6sgbymszspybts.jpg',
    review: 'Nice'
  },
  createBookErrorData: {
    author: 'a',
    category: 'a',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1511133007/g2exfx6sgbymszspybts.jpg',
    review: 'Nice'
  },
  createBookErrorResponse: 'An error occured',
  createBookResponse: {
    message: 'Succesfully added'
  },
  getABookData: {
    id: 7,
    title: 'Rage of Angels',
    author: 'Sidney Sheldon',
    category: 'Fiction',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1513096198/prbdr6eghnnrcewwpodf.jpg',
    review: 'Nice Book',
    createdAt: '2017-12-12T11:11:10.823Z',
    updatedAt: '2017-12-12T16:30:00.352Z',
    userId: null,
    categoryId: null
  },
  getAllBooksData: {
    id: 7,
    title: 'Rage of Angels',
    author: 'Sidney Sheldon',
    category: 'Fiction',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1513096198/prbdr6eghnnrcewwpodf.jpg',
    review: 'Nice Book',
    createdAt: '2017-12-12T11:11:10.823Z',
    updatedAt: '2017-12-12T16:30:00.352Z',
    userId: null,
    categoryId: null
  },
  getAllBorrowedBooksData: {
    id: 7,
    title: 'Rage of Angels',
    author: 'Sidney Sheldon',
    category: 'Fiction',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1513096198/prbdr6eghnnrcewwpodf.jpg',
    review: 'Nice Book',
    createdAt: '2017-12-12T11:11:10.823Z',
    updatedAt: '2017-12-12T16:30:00.352Z',
    userId: null,
    categoryId: null
  },
  getCategoryData: {
    category: 'Fiction'
  },
  getUserData: {
    firstname: 'daddy',
    lastname: 'daddy',
    username: 'daddy',
    password: 'daddy',
    email: 'daddy@gmail.com'
  },
  bookData: {
    title: 'a',
    author: 'b',
    category: 'a',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1511133007/g2exfx6sgbymszspybts.jpg',
    review: 'Nice'
  },
  returnBookData: {
    message: 'Successfully Returned'
  },
  userData: {
    verifyPassword: '1234',
    password: '4321'
  },
  errorMessage: 'An error occurred',
  image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1515403427/ena7dhy6xriztom6vuvo.jpg'
};

export default mockData;
