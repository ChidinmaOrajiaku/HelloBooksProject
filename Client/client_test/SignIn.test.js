import configureMockStore from 'redux-mock-store';
import jwt from 'jsonwebtoken';
import thunk from 'redux-thunk';
import moxios from 'moxios';
import mockData from './mocks/mockData';
import * as SigninActions from '../actions/signinAction';
import * as GetUserActions from '../actions/getUser';
import * as ModifyBookActions from '../actions/modifyBooks';
import * as GetUserBorrowedActions from '../actions/getUserBorrowedBooks';
import * as CreateBookActions from '../actions/createBooks';
import * as GetBookActions from '../actions/getABook';
import * as GetAllBooksActions from '../actions/getAllBooks';
import * as GetAllBorrowedBooksActions from '../actions/getAllBorrowedBooks';
import * as DeleteBookActions from '../actions/deleteBooks';
import * as BorrowBookActions from '../actions/borrowBooks';
import * as CategoryActions from '../actions/getCategory';
import * as CountUsersActions from '../actions/adminCountUsers';
import * as ActionTypes from '../actions/types';

const middlewares = [thunk];

const mockStore = configureMockStore(middlewares);

/**
 * Thunk test
 */
describe('SignIn actions', () => {
  beforeEach(() => {
    moxios.install();
  });
  afterEach(() => {
    moxios.uninstall();
  });

  it('creates SET_CURRENT_USER when user signs in', async (done) => {
    jest.setTimeout(10000) 
    const { userDetails, userDetailsResponse } = mockData;
    moxios.stubRequest('/api/v1/users/signin', {
      status: 200,
      response: userDetails
    });
    const expectedActions = [{ 
        type: ActionTypes.SET_CURRENT_USER, 
        user: userDetailsResponse 
    }];
    const store = mockStore({});
    await store.dispatch(SigninActions.userSigninRequest(userDetails))
      .then(() => {
        expect(store.getActions()).toEqual(expectedActions);
      });
      done();
  });
});

describe('Count User actions', () => {
  beforeEach(() => {
    moxios.install();
  });
  afterEach(() => {
    moxios.uninstall();
  });

  it('gets COUNT_USERS when admin counts users', async (done) => {
    jest.setTimeout(10000) 
    const { userCountResponse } = mockData;
    moxios.stubRequest('/api/v1/users/', {
      status: 200,
      response: userCountResponse
    });
    const expectedActions = [{ 
        type: ActionTypes.COUNT_USERS, 
        adminCountUsers: 7
    }];
    const store = mockStore({});
    await store.dispatch(CountUsersActions.adminCountUserRequest())
      .then(() => {
        expect(store.getActions()).toEqual(expectedActions);
      });
      done();
  });
});

describe('Borrow books actions', () => {
    beforeEach(() => {
      moxios.install();
    });
    afterEach(() => {
      moxios.uninstall();
    });
  
    it('creates BORROW_BOOKS_SUCCESSFUL user borrows book', async (done) => {
      jest.setTimeout(10000) 
      const { borrowBookData, borrowBookDataResponse } = mockData;
      moxios.stubRequest(`/api/v1/users/${borrowBookData.userId}/books`, {
        status: 201,
        response: borrowBookDataResponse
      });
      const expectedActions = [{ 
          type: ActionTypes.BORROW_BOOKS_SUCCESSFUL, 
          response: borrowBookDataResponse
      }];
      const store = mockStore({});
      await store.dispatch(BorrowBookActions.borrowRequest(borrowBookData.bookId))
        .then(() => {
          expect(store.getActions()).toEqual(expectedActions);
        });
        done();
    });
    it('creates BORROW_BOOKS_FAILED user borrows book', async (done) => {
        jest.setTimeout(10000) 
        const { borrowBookData} = mockData;
        moxios.stubRequest(`/api/v1/users/${borrowBookData.userId}/books`, {
          status: 400,
          error: "An error occurred"
        });
        const expectedActions = [{ 
            type: ActionTypes.BORROW_BOOKS_FAILED, 
            error: "An error occurred"
        }];
        const store = mockStore({});
        await store.dispatch(BorrowBookActions.borrowRequest(borrowBookData.bookId))
          .then(() => {
            expect(store.getActions()).toEqual(expectedActions);
          });
          done();
      });
  });

  describe('Create books actions', () => {
    beforeEach(() => {
      moxios.install();
    });
    afterEach(() => {
      moxios.uninstall();
    });
  
    it('creates CREATE_BOOKS_SUCCESSFUL book', async (done) => {
      jest.setTimeout(10000) 
      const { createBookData, createBookResponse } = mockData;
      moxios.stubRequest('/api/v1/users/books', {
        status: 201,
        response: createBookResponse
      });
      const expectedActions = [{ 
          type: ActionTypes.CREATE_BOOKS_SUCCESSFUL, 
          response: createBookResponse
      }];
      const store = mockStore({});
      await store.dispatch(CreateBookActions.adminAddRequest(createBookData))
        .then(() => {
          expect(store.getActions()).toEqual(expectedActions);
        });
        done();
    });
    it('creates CREATE_BOOKS_FAILED book', async (done) => {
        jest.setTimeout(10000) 
        const { createBookData} = mockData;
        moxios.stubRequest('/api/v1/users/books', {
          status: 400,
          error: "An error occurred"
        });
        const expectedActions = [{ 
            type: ActionTypes.CREATE_BOOKS_FAILED, 
            error: "An error occurred"
        }];
        const store = mockStore({});
        await store.dispatch(CreateBookActions.adminAddRequest(createBookData))
          .then(() => {
            expect(store.getActions()).toEqual(expectedActions);
          });
          done();
      });
  });

  describe('Get a book actions', () => {
    beforeEach(() => {
      moxios.install();
    });
    afterEach(() => {
      moxios.uninstall();
    });
  
    it('get a GET_A_BOOK_SUCCESSFUL book', async (done) => {
      jest.setTimeout(10000) 
      const { getABookData } = mockData;
      moxios.stubRequest('/api/v1/books/1', {
        status: 200,
        response: getABookData
      });
      const expectedActions = [{ 
          type: ActionTypes.GET_A_BOOK_SUCCESSFUL, 
          response: getABookData
      }];
      const store = mockStore({});
      await store.dispatch(GetBookActions.getBookRequest(1))
        .then(() => {
          expect(store.getActions()).toEqual(expectedActions);
        });
        done();
    });
    ('get a GET_A_BOOK_FAILED book', async (done) => {
        jest.setTimeout(10000) 
        moxios.stubRequest('/api/v1/books/1', {
          status: 400,
          error: "An error occurred"
        });
        const expectedActions = [{ 
            type: ActionTypes.GET_A_BOOK_FAILED, 
            error: "An error occurred"
        }];
        const store = mockStore({});
        await store.dispatch(GetBookActions.getBookRequest())
          .then(() => {
            expect(store.getActions()).toEqual(expectedActions);
          });
          done();
      });
  });

  describe('Delete a book actions', () => {
    beforeEach(() => {
      moxios.install();
    });
    afterEach(() => {
      moxios.uninstall();
    });
  
    it('delete a DELETE_BOOKS_SUCCESSFUL book', async (done) => {
      jest.setTimeout(10000) 
      moxios.stubRequest('/api/v1/books/1', {
        status: 200,
        response: []
      });
      const expectedActions = [{ 
          type: ActionTypes.DELETE_BOOKS_SUCCESSFUL, 
          response: []
      }];
      const store = mockStore({});
      await store.dispatch(DeleteBookActions.adminDeleteRequest(1))
        .then(() => {
          expect(store.getActions()).toEqual(expectedActions);
        });
        done();
    });
    it('delete a DELETE_BOOKS_FAILED book', async (done) => {
        jest.setTimeout(10000) 
        moxios.stubRequest('/api/v1/books/1', {
          status: 400,
          error: "An error occurred"
        });
        const expectedActions = [{ 
            type: ActionTypes.DELETE_BOOKS_FAILED, 
            error: "An error occurred"
        }];
        const store = mockStore({});
        await store.dispatch(DeleteBookActions.adminDeleteRequest(1))
          .then(() => {
            expect(store.getActions()).toEqual(expectedActions);
          });
          done();
      });
  });

  describe('Get all books actions', () => {
    beforeEach(() => {
      moxios.install();
    });
    afterEach(() => {
      moxios.uninstall();
    });
  
    it('get all GET_BOOKS_SUCCESSFUL books', async (done) => {
      jest.setTimeout(10000) 
      const { getAllBooksData } = mockData;
      moxios.stubRequest('/api/v1/users/books', {
        status: 200,
        response: getAllBooksData
      });
      const expectedActions = [{ 
          type: ActionTypes.GET_BOOKS_SUCCESSFUL, 
          response: getAllBooksData
      }];
      const store = mockStore({});
      await store.dispatch(GetAllBooksActions.getRequest())
        .then(() => {
          expect(store.getActions()).toEqual(expectedActions);
        });
        done();
    });
    it('get all GET_BOOKS_FAILED books', async (done) => {
        jest.setTimeout(10000) 
        moxios.stubRequest('/api/v1/users/books', {
          status: 400,
          error: "An error occurred"
        });
        const expectedActions = [{ 
            type: ActionTypes.GET_BOOKS_FAILED, 
            error: "An error occurred"
        }];
        const store = mockStore({});
        await store.dispatch(GetAllBooksActions.getRequest(1))
          .then(() => {
            expect(store.getActions()).toEqual(expectedActions);
          });
          done();
      });
});
 describe('Get all borrowed books actions', () => {
    beforeEach(() => {
        moxios.install();
    });
    afterEach(() => {
        moxios.uninstall();
    });

    it('get all GET_BORROWED_BOOKS_SUCCESSFUL books', async (done) => {
        jest.setTimeout(10000) 
        const { getAllBorrowedBooksData } = mockData;
        moxios.stubRequest('/api/v1/users/books/unreturned', {
        status: 200,
        response: getAllBorrowedBooksData
        });
        const expectedActions = [{ 
            type: ActionTypes.GET_BORROWED_BOOKS_SUCCESSFUL, 
            response: getAllBorrowedBooksData
        }];
        const store = mockStore({});
        await store.dispatch(GetAllBorrowedBooksActions.admingetBorrowedRequest())
        .then(() => {
            expect(store.getActions()).toEqual(expectedActions);
        });
        done();
    });
    it('get all GET_BORROWED_BOOKS_FAILED books', async (done) => {
        jest.setTimeout(10000) 
        const { getAllBorrowedBooksData } = mockData;
        moxios.stubRequest('/api/v1/users/books/unreturned', {
            status: 400,
            error: "An error occurred"
        });
        const expectedActions = [{ 
            type: ActionTypes.GET_BORROWED_BOOKS_FAILED, 
            error: "An error occurred"
        }];
        const store = mockStore({});
        await store.dispatch(GetAllBorrowedBooksActions.admingetBorrowedRequest())
            .then(() => {
            expect(store.getActions()).toEqual(expectedActions);
            });
            done();
        });
  });

  describe('Get category actions', () => {
    beforeEach(() => {
        moxios.install();
    });
    afterEach(() => {
        moxios.uninstall();
    });

    it('get all GET_CATEGORY_SUCCESSFUL category', async (done) => {
        jest.setTimeout(10000) 
        const { getCategoryData } = mockData;
        moxios.stubRequest('/api/v1/books/category/all', {
        status: 200,
        response: getCategoryData
        });
        const expectedActions = [{ 
            type: ActionTypes.GET_CATEGORY_SUCCESSFUL, 
            response: getCategoryData
        }];
        const store = mockStore({});
        await store.dispatch(CategoryActions.getAllCategoryRequest())
        .then(() => {
            expect(store.getActions()).toEqual(expectedActions);
        });
        done();
    });
    it('get all GET_CATEGORY_FAILED category', async (done) => {
        jest.setTimeout(10000) 
        moxios.stubRequest('/api/v1/books/category/all', {
            status: 400,
            error: "An error occurred"
        });
        const expectedActions = [{ 
            type: ActionTypes.GET_CATEGORY_FAILED, 
            error: "An error occurred"
        }];
        const store = mockStore({});
        await store.dispatch(CategoryActions.getAllCategoryRequest())
            .then(() => {
            expect(store.getActions()).toEqual(expectedActions);
            });
            done();
        });
  });
  describe('Get user actions', () => {
    beforeEach(() => {
        moxios.install();
    });
    afterEach(() => {
        moxios.uninstall();
    });

    it('get GET_USER_SUCCESSFUL user details', async (done) => {
        jest.setTimeout(10000) 
        const { getUserData } = mockData;
        moxios.stubRequest('/api/v1/users/1', {
        status: 200,
        response: getUserData
        });
        const expectedActions = [{ 
            type: ActionTypes.GET_USER_SUCCESSFUL, 
            response: getUserData
        }];
        const store = mockStore({});
        await store.dispatch(GetUserActions.getUserDataRequest(1))
        .then(() => {
            expect(store.getActions()).toEqual(expectedActions);
        });
        done();
    });
    it('get GET_USER_FAILED user details', async (done) => {
        jest.setTimeout(10000) 
        moxios.stubRequest('/api/v1/users/1', {
            status: 400,
            error: "An error occurred"
        });
        const expectedActions = [{ 
            type: ActionTypes.GET_USER_FAILED, 
            error: "An error occured"
        }];
        const store = mockStore({});
        await store.dispatch(GetUserActions.getUserDataRequest(1))
            .then(() => {
            expect(store.getActions()).toEqual(expectedActions);
            });
            done();
        });
  });

  describe('Get user borrowed books actions', () => {
    beforeEach(() => {
        moxios.install();
    });
    afterEach(() => {
        moxios.uninstall();
    });

    it('get GET_USER_BORROWED_SUCCESSFUL user borrowed books details', async (done) => {
        jest.setTimeout(10000) 
        const { getAllBorrowedBooksData } = mockData;
        moxios.stubRequest('/api/v1/users/1/history', {
        status: 200,
        response: getAllBorrowedBooksData
        });
        const expectedActions = [{ 
            type: ActionTypes.GET_USER_BORROWED_SUCCESSFUL, 
            response: getAllBorrowedBooksData
        }];
        const store = mockStore({});
        await store.dispatch(GetUserBorrowedActions.getUserBorrowed(1))
        .then(() => {
            expect(store.getActions()).toEqual(expectedActions);
        });
        done();
    });
    it('get GET_USER_BORROWED_FAILED user details', async (done) => {
        jest.setTimeout(10000) 
        moxios.stubRequest('/api/v1/users/1/history', {
            status: 400,
            error: "An error occurred"
        });
        const expectedActions = [{ 
            type: ActionTypes.GET_USER_BORROWED_FAILED, 
            error: "An error occured"
        }];
        const store = mockStore({});
        await store.dispatch(GetUserBorrowedActions.getUserBorrowed(1))
            .then(() => {
            expect(store.getActions()).toEqual(expectedActions);
            });
            done();
        });
  });

  describe('Modify books actions', () => {
    beforeEach(() => {
        moxios.install();
    });
    afterEach(() => {
        moxios.uninstall();
    });

    it('modify MODIFY_BOOKS_SUCCESSFUL books', async (done) => {
        jest.setTimeout(10000) 
        const { bookData } = mockData;
        moxios.stubRequest('/api/v1/books/1', {
        status: 201,
        response: bookData
        });
        const expectedActions = [{ 
            type: ActionTypes.MODIFY_BOOKS_SUCCESSFUL, 
            response: bookData
        }];
        const store = mockStore({});
        await store.dispatch(ModifyBookActions.adminModifyRequest(1))
        .then(() => {
            expect(store.getActions()).toEqual(expectedActions);
        });
        done();
    });
    it("modify MODIFY_BOOKS_FAILED books actions", async (done) => {
        jest.setTimeout(10000) 
        moxios.stubRequest('/api/v1/books/1', {
            status: 400,
            error: "An error occurred"
        });
        const expectedActions = [{ 
            type: ActionTypes.MODIFY_BOOKS_FAILED, 
            error: "An error occured"
        }];
        const store = mockStore({});
        await store.dispatch(ModifyBookActions.adminModifyRequest(1))
            .then(() => {
            expect(store.getActions()).toEqual(expectedActions);
            });
            done();
        });
  });