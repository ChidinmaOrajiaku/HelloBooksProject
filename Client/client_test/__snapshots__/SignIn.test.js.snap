// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> should match snapshot test 1`] = `
<div
  className="signIn"
>
  <div
    className="bodyNavigation"
  >
     
    <Component />
     
  </div>
  <div
    className="row container"
  >
    <div
      className="col s12 m6 offset-m3"
    >
      <div
        className="card"
        data-aos="flip-up"
      >
        <div
          className="card-content black-text"
        >
          <form
            id="form"
            onSubmit={[Function]}
          >
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <input
                  className="validate"
                  id="email"
                  onChange={[Function]}
                  required="required"
                  type="email"
                  value=""
                />
                <label
                  htmlFor="email"
                >
                  Email
                </label>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <input
                  className="validate"
                  id="password"
                  onChange={[Function]}
                  required="required"
                  type="password"
                  value=""
                />
                <label
                  htmlFor="password"
                >
                  Password
                </label>
              </div>
            </div>
            <button
              className="btn waves-effect waves-light"
              name="action"
              type="submit"
            >
              Sign In
              <i
                className="material-icons right"
              >
                send
              </i>
            </button>
          </form>
        </div>
        <div
          className="account"
        >
          <p>
             Not a member?
            <Link
              replace={false}
              to="/register"
            >
              Sign Up
            </Link>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
